addons/test_mail/tests/test_attachment.py

```diff
@@ -0,0 +1,72 @@
+from odoo.addons.test_mail.tests.common import TestMailCommon
+from odoo.exceptions import AccessError
+
+class TestAttachment(TestMailCommon):
+
+    def test_attachment_forbid_unlink(self):
+        """Check that removing message attachments is prevented on other user's messages."""
+        def write(attachment, w_user, w_sudo=False):
+            attachment.with_user(w_user).sudo(w_sudo).write({'name': f'{w_user.name} {attachment.name[-1]}'})
+
+        def unlink(attachment, w_user, w_sudo=False):
+            attachment.exists().with_user(w_user).sudo(w_sudo).unlink()
```

Comment:
Why this exists ?

Reply:
So I can reference it in a loop for testing :p